src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private static final Random random = new Random();
+    private static final double MAX_SIZE = 100.0;
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Color color = colorSupplier.getRandomColor();
+
+        switch (random.nextInt(5)) {
+            case 0:
+                return new Square(color, random.nextDouble() * MAX_SIZE);
```

Comment:
For better readability, consider creating separate variables for the random values before passing them to the constructor. This aligns with the checklist recommendation to not pass expressions directly in constructors.